### PR TITLE
Update jetpack-connect-plans-grid

### PR DESCRIPTION
**Pull request by sirreal**
* Pass linter (es6 class, localize)
* Add propTypes

Lint fixes required for: Separate jetpack-connect from signup #12991

To test:

* Use this branch.
* Go to `/jetpack/connect`.
* Enter your site's URL (disconnected Jetpack site with no plans).
* Alternatively `/jetpack/connect/plans/:siteslug` may work under some conditions.
* Test buttons, render etc.
* Ensure there are no regressions. Watch the console (preserve log) for any errors/warnings.

### CODE REVIEW

File: client/signup/jetpack-connect/plans-grid.jsx [hunk 1]
```diff
@@ -10,22 +11,38 @@ import Main from 'components/main';
 import StepHeader from '../step-header';
 import PlansFeaturesMain from 'my-sites/plans-features-main';
 
-export default React.createClass( {
-	displayName: 'JetpackPlansGrid',
+class JetpackPlansGrid extends Component {
+	static propTypes = {
+		basePlansPath: PropTypes.string.isRequired,
```
Comment (tyxla) [inline client/signup/jetpack-connect/plans-grid.jsx]: Why required? `PlansFeaturesMain` appears to not require `basePlansPath` to be required.

File: client/signup/jetpack-connect/plans-grid.jsx [hunk 2]
```diff
@@ -10,22 +11,38 @@ import Main from 'components/main';
...
+		hideFreePlan: PropTypes.bool,
```
Comment (tyxla) [inline client/signup/jetpack-connect/plans-grid.jsx]: Can we also try to define some `defaultProps` for all non-required propTypes?

File: client/signup/jetpack-connect/plans-grid.jsx (hunk 1 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-grid.jsx]: 👍  18ddb79f7

File: client/signup/jetpack-connect/plans-grid.jsx (hunk 2 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-grid.jsx]: @tyxla I think that's worthy of another PR, there are a few small but significant changes that can be made related to defaults. I'm going to handle that separately.

**Comment by johnHackworth** [comment]
looks good to me too :+1:

**Pull request merged**